core/src/main/java/hudson/util/FormValidation.java

```diff
@@ -372,6 +373,7 @@ public static FormValidation validateExecutable(String exe, FileValidator exeVal
                 File fexe = new File(dir,exe+".exe");
                 if(fexe.exists())   return exeValidator.validate(fexe);
             }
+            tokenizedPath = tokenizedPathBuilder.toString();
 
             tokenizedPath += ".";
```

Comment:
This should be appended before the toString() call.

Reply:
Oops, I fixed that locally.

Reply:
Oh. I think I know what happened. I fixed it locally but collided w/ accepted recommended changes from daniel-beck ... I'll restore the fix...